### ISSUE TITLE
unclear behaviour of imcfolderwriter.write_imc_folder()

### ISSUE THREAD

**Issue by smaffiol** [BodenmillerGroup/imctools#39]
imctools.io.imcfolderwriter.write_imc_folder(self, zipfolder=True, remove_folder=None)

remove_folder is used at the end of the method to cleanup out_folder - the folder where all the data have just been created:         
if remove_folder:
     os.removedirs(out_folder)

While this makes sense when also zipfolder=True (which is the case when also zipfolder=True and remove_folder=None) - I fail to see the logic when zipfolder=False and  remove_folder=True
in this case, one would just remove everything just created

did I read it correctly ?
S.

**Comment by votti** [comment]
Yes, that is the correct behavior. What more `correct` behavior would you expect if somebody purposely sets these arguments in this manner (zipfolder=False and remove_folder=True)?

**Comment by smaffiol** [comment]
remove_folder should be a consequence of the zipfolder.
So, my suggestion:
- remove folder as a default operation if zipfolder = True
- add an option 'keep_folder' = False (to check whether the original folder
should be maintained once zipped)
check 'keep_folder' only if zipfolder = True
in this case:
- zipfolder = False and keep_folder = True has no effect
- zipfolder = False and keep_folder = False, in this case keep_folder is
ignored

make sense ?

S.


Il giorno gio 20 dic 2018 alle ore 09:35 Vito Zanotelli <
notifications@github.com> ha scritto:
